test/parallel/test-inspector-network-resource.js

```diff
@@ -0,0 +1,126 @@
+// Flags: --inspect=0 --experimental-network-inspection
+'use strict';
+const common = require('../common');
+
+common.skipIfInspectorDisabled();
+
+const { NodeInstance } = require('../common/inspector-helper');
+const test = require('node:test');
+const assert = require('node:assert');
+const path = require('path');
+const fs = require('fs');
+
+const resourceUrl = 'http://localhost:3000/app.js';
+const resourcePath = path.join(__dirname, '../fixtures/inspector-network-resource/app.js.map');
+
+const script = `
+  const { NetworkResources } = require('node:inspector');
+  const fs = require('fs');
+  NetworkResources.put('${resourceUrl}', fs.readFileSync('${resourcePath.replace(/\\/g, '\\').replace(/'/g, "\\'")}', 'utf8'));
+  console.log('Network resource loaded:', '${resourceUrl}');
```

Comment:
`setupSessionAndPauseAtEvalLine4` can be simplied with a `inspector.waitForDebugger()` call (or with `--inspect-brk` and a `debugger` statement, without setting breakpoints on the fly.

```suggestion
  console.log('Network resource loaded:', '${resourceUrl}');
  debugger;
```